### ISSUE TITLE
[GENERAL SUPPORT]: Clarification on GPEI Model Internal Parameters (surrogate model & Acquisition function)

### ISSUE THREAD

**Issue by vinaysaini94** [facebook/Ax#2904]
### Question

Clarification on GPEI Model Parameters

```shell
Hi Community,

I need a bit of clarification regarding the following:
(with Ax version V0.4.3 before maintenance release)
When we use:


Models.GPEI(
    experiment=experiment,
    data=data
)


What parameters are internally selected, such as the surrogate model (e.g., `model_class`, `covar_module_class`, kernel) and acquisition function class (`acqf_class`)? Is this equivalent to explicitly selecting the `SingleTaskGP` in BoTorch, as shown below? ((with Botorch version v0.12.0)

```python
Models.BOTORCH_MODULAR(
    experiment=experiment,
    data=data,
    torch_device=tkwargs["device"],
    torch_dtype=tkwargs["dtype"],
    surrogate=Surrogate(
        botorch_model_class=SingleTaskGP,
        covar_module_class=RBFKernel,
        covar_module_options={"lengthscale_prior": lengthscale_prior, 
                              "lengthscale_constraint": lengthscale_constraint},
        likelihood_options={"noise_prior": noise_prior},
    ), 
    botorch_acqf_class=qLogExpectedImprovement
)
```

Any detailed information and clarification in this regard would be really helpful.

Thanks!
```


### Code of Conduct

- [X] I agree to follow this Ax's Code of Conduct

**Comment by bernardbeckerman** [comment]
> Is this equivalent to explicitly selecting the `SingleTaskGP` in BoTorch, as shown below?

Yes. Please see our docs for more details on defaults (link: https://ax.dev/tutorials/modular_botax.html#Appendix-2:-Default-surrogate-models-and-acquisition-functions).

**Comment by vinaysaini94** [comment]
Hi @bernardbeckerman,

Thanks for the link! I took a look, but I’m still a bit unclear on which specific defaults for the surrogate model and acquisition function are applied when using `Models.GPEI(...)`. Does it default to `SingleTaskGP` with an `RBFKernel` like in the `Models.BOTORCH_MODULAR` example I shared? Also, which acquisition function is used by default (e.g., `qLogExpectedImprovement`)? Any additional clarification would be great!

I’m still having a little confusion about the internal parameter selection when I use the `GPEI` model.

Here’s a snippet of my code:

```python
from ax.modelbridge.registry import Models
model = Models.GPEI(experiment=experiment, data=data)
```

What exactly are the parameters selected (e.g., surrogate model, acquisition function, lengthscale_prior, outputscale_prior)? Is there a way to print the full list of these parameters?

Additionally, I’d like to confirm whether the `SingleTaskGP` is selected as the surrogate model by default. If so, does it take into account the "Dimension-scaled Lengthscale Prior" modification in the current Ax version (v0.4.3), or is that feature only available in the BoTorch library?

Thanks in advance for your help!

**Comment by saitcakmak** [comment]
Hi @vinaysaini94. `Models.GPEI` is deprecated and we do not recommend using it. It will be fully removed in the near future. I'd recommend switching to `Models.BoTorch` instead, which uses (in most cases) `SingleTaskGP` (with BoTorch default RBF kernel + dim-scaled priors) along with `qLogNoisyExpectedImprovement` or `qLogNoisyExpectedHypervolumeImprovement` acquisition functions depending on whether it is a multi-objective or a single-objective problem.

Clarification on "in most cases": If there are fidelity or task parameters in the search space, a `MultiTaskGP` or `SingleTaskMultiFidelityGP` may be used.

> Additionally, I’d like to confirm whether the SingleTaskGP is selected as the surrogate model by default. If so, does it take into account the "Dimension-scaled Lengthscale Prior" modification in the current Ax version (v0.4.3), or is that feature only available in the BoTorch library?

Ax 0.4.3 utilizes the new kernel & priors from BoTorch 0.12.0.
